### PR TITLE
Allow to realy NEVER check for updates, even on first start.

### PR DESCRIPTION
**Pull request by sdedic**
AutoUpdate can be configured to check never for updates, but regardless of that setting, the check happens on 1st startup. For vscode distribution, I would like to ensure the autoupdate really never connects to AU center. At the same time I did not want to affect the NB operation - so I've added a 'branding API' that can be used to disable the first forced check. nbls distribution then contains the appropriate branding that disables it.

Maybe a command will be added later to faciliate with AU to install updates, but given how the vscode Marketplace works, it may not be necessary.

### CODE REVIEW

File: platform/autoupdate.ui/src/org/netbeans/modules/autoupdate/ui/actions/AutoupdateCheckScheduler.java [hunk 1]
```diff
@@ -67,6 +69,12 @@
  * @author Jiri Rechtacek
  */
 public class AutoupdateCheckScheduler {
+    /**
+     * System property that allows to skip forced initial check. If set to true, the check is not forced, if a check was never
+     * was performed - the scheduling is always driven by AU settings.
+     */
+    private static final String INITIAL_CHECK_SKIP_OPT = "autoupdate.initial.check.skip"; // NOI18N
```
Comment (dbalek) [inline platform/autoupdate.ui/src/org/netbeans/modules/autoupdate/ui/actions/AutoupdateCheckScheduler.java:76]: Is it used somewhere?

**Review by dbalek — APPROVED** [review]

File: platform/autoupdate.ui/src/org/netbeans/modules/autoupdate/ui/actions/AutoupdateCheckScheduler.java (hunk 1 above)
Comment (sdedic) [reply platform/autoupdate.ui/src/org/netbeans/modules/autoupdate/ui/actions/AutoupdateCheckScheduler.java]: Fixed in 56895dcc5a8b

**Comment by sdedic** [comment]
@neilcsmith-net  re bug: I thought that it is a bug initially .... but in fact the current behaviour is (with NEVER) to check just once in a completely empty userdir - and then never again. This could be actually beneficial, as it fetches info for a fresh installation. I did not want to interfere with the behaviour.

Do you think the behaviour is really buggy ? If so then clearly simple fix could be better.

Re. removing from vscode distro -- it's not that easy, as `autoupdate.ui` is referenced from `java.source` (for downloading nbjavac), from `org.netbeans.modules.junit` (downloading junit libraries I think), `maven` (downloads apisupport for NBM packaging) and `code.analysis` (code analysis plugins download)

**Comment by neilcsmith-net** [comment]
How is an empty userdir going to be set to `NEVER` though?  Either it's an imported setting from the user, which should be respected, or it's a branded default of the application, which should also be respected.  Or am I missing a scenario?

> Re. removing from vscode distro -- it's not that easy, as `autoupdate.ui` is referenced from `java.source` (for downloading nbjavac), from `org.netbeans.modules.junit` (downloading junit libraries I think), `maven` (downloads apisupport for NBM packaging) and `code.analysis` (code analysis plugins download)

I'm aware.  I have the same issue in my platform application, which uses `java.source` etc. and has no need for the update centre checks or UI either.  I meant it's a shame there's a hard dependency there.

**Comment by neilcsmith-net** [comment]
Yes, but I don't think that changes the argument.  How can this ever be set to `NEVER` unless explicitly set?  And if explicitly set, it seems a bug to ever connect without explicit user action.

Incidentally, the other bundle needs a license to pass RAT.  If it's staying in at all of course! :smile:

**Comment by sdedic** [comment]
> Yes, but I don't think that changes the argument. How can this ever be set to `NEVER` unless explicitly set? And if explicitly set, it seems a bug to ever connect without explicit user action.
> 

For example, vscode *brands* the default setting to NEVER. So it is not explicitly set (by the user). It will be also implicitly set by **importing** old settings by a new installation - and this is the case, where it *might* be valid to do one-time fetch.

How to reach a consensus, or a confirmation of the 'bug suspection' here ?

**Comment by neilcsmith-net** [comment]
> For example, vscode _brands_ the default setting to NEVER. So it is not explicitly set (by the user). It will be also implicitly set by **importing** old settings by a new installation - and this is the case, where it _might_ be valid to do one-time fetch.

If it's branded, it's been explicitly set by the application developer, if imported settings it has been explicitly set at some point by the user.  In either case, it seems wrong to ever connect without manual interaction.

This is still marked for NB18.  Given the other changes have addressed immediate concerns in VSCode, do you want to adapt this PR for NB18 or NB19, or close it?

**Comment by neilcsmith-net** [comment]
@sdedic are we closing this?

**Pull request closed without merging**